pkg/apis/capabilities/v1beta1/backend_types.go

```diff
@@ -0,0 +1,307 @@
+package v1beta1
+
+import (
+	"reflect"
+	"regexp"
+
+	"github.com/3scale/3scale-operator/pkg/common"
+
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+const (
+	BackendKind = "Backend"
+
+	// BackendInvalidConditionType represents that the combination of configuration
+	// in the BackendSpec is not supported. This is not a transient error, but
+	// indicates a state that must be fixed before progress can be made.
+	// Example: the spec references non existing internal Metric reference
+	BackendInvalidConditionType common.ConditionType = "Invalid"
+
+	// BackendSyncedConditionType indicates the product has been successfully synchronized.
+	// Steady state
+	BackendSyncedConditionType common.ConditionType = "Synced"
+
+	// BackendFailedConditionType indicates that an error occurred during synchronization.
+	// The operator will retry.
+	BackendFailedConditionType common.ConditionType = "Failed"
+)
+
+var (
+	//
+	backendSystemNameRegexp = regexp.MustCompile("[^a-zA-Z0-9]+")
+)
+
+// ProductStatusError represents that the combination of configuration in the BackendSpec
+// is not supported by this cluster. This is not a transient error, but
+// indicates a state that must be fixed before progress can be made.
+// Example: the BackendSpec references non existing internal Metric refenrece
+type BackendStatusError string
+
+// BackendSpec defines the desired state of Backend
+type BackendSpec struct {
+	// Name is human readable name for the backend
+	Name string `json:"name"`
+
+	// SystemName identifies uniquely the product within the account provider
+	// Default value will be sanitized Name
+	// +optional
+	SystemName string `json:"systemName,omitempty"`
+
+	// +kubebuilder:validation:Pattern=`^https?:\/\/.*$`
+	PrivateBaseURL string `json:"privateBaseURL,omitempty"`
+
+	// +optional
+	Description string `json:"description,omitempty"`
+
+	// +optional
+	MappingRules []MappingRuleSpec `json:"mappingRules,omitempty"`
+
+	// Metrics
+	// Map: system_name -> MetricSpec
+	// system_name attr is unique for all metrics AND methods
+	// In other words, if metric's system_name is A, there is no metric or method with system_name A.
+	// +optional
+	Metrics map[string]MetricSpec `json:"metrics,omitempty"`
+
+	// Methods
+	// Map: system_name -> MethodSpec
+	// system_name attr is unique for all metrics AND methods
+	// In other words, if metric's system_name is A, there is no metric or method with system_name A.
+	// +optional
+	Methods map[string]Methodpec `json:"methods,omitempty"`
```

Comment:
Typo in the struct name. I guess it should be MethodSpec

Reply:
fixed